From 2adef6e48d84bb65b2717dced64716d9566e1d68 Mon Sep 17 00:00:00 2001
From: Jay Logue <jaylogue@google.com>
Date: Fri, 30 Aug 2019 09:53:22 -0700
Subject: [PATCH] openweave-esp32-lwip-patch-0010 : Make nd6_select_router()
 public

Allow nd6_select_router() function to be called by a user of LwIP.  This
provides a means for user code to determine if a default route exists on
a particular interface.
---
 lwip/src/core/ipv6/nd6.c    | 3 +--
 lwip/src/include/lwip/nd6.h | 1 +
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lwip/src/core/ipv6/nd6.c b/lwip/src/core/ipv6/nd6.c
index 0b03ee8..91f07fe 100644
--- a/lwip/src/core/ipv6/nd6.c
+++ b/lwip/src/core/ipv6/nd6.c
@@ -99,7 +99,6 @@ static void nd6_free_neighbor_cache_entry(s8_t i);
 static s8_t nd6_find_destination_cache_entry(const ip6_addr_t *ip6addr);
 static s8_t nd6_new_destination_cache_entry(void);
 static s8_t nd6_is_prefix_in_netif(const ip6_addr_t *ip6addr, struct netif *netif);
-static s8_t nd6_select_router(const ip6_addr_t *ip6addr, struct netif *netif);
 static s8_t nd6_get_router(const ip6_addr_t *router_addr, struct netif *netif);
 static s8_t nd6_new_router(const ip6_addr_t *router_addr, struct netif *netif);
 static s8_t nd6_get_onlink_prefix(ip6_addr_t *prefix, struct netif *netif);
@@ -1407,7 +1406,7 @@ nd6_is_prefix_in_netif(const ip6_addr_t *ip6addr, struct netif *netif)
  * @return the default router entry index, or -1 if no suitable
  *         router is found
  */
-static s8_t
+s8_t
 nd6_select_router(const ip6_addr_t *ip6addr, struct netif *netif)
 {
   s8_t i;
diff --git a/lwip/src/include/lwip/nd6.h b/lwip/src/include/lwip/nd6.h
index d9fba97..cb023bc 100644
--- a/lwip/src/include/lwip/nd6.h
+++ b/lwip/src/include/lwip/nd6.h
@@ -65,6 +65,7 @@ void nd6_tmr(void);
 void nd6_input(struct pbuf *p, struct netif *inp);
 void nd6_clear_destination_cache(void);
 struct netif *nd6_find_route(const ip6_addr_t *ip6addr);
+s8_t nd6_select_router(const ip6_addr_t * ip6addr, struct netif * netif);
 err_t nd6_get_next_hop_addr_or_queue(struct netif *netif, struct pbuf *q, const ip6_addr_t *ip6addr, const u8_t **hwaddrp);
 u16_t nd6_get_destination_mtu(const ip6_addr_t *ip6addr, struct netif *netif);
 #if LWIP_ND6_TCP_REACHABILITY_HINTS
-- 
2.17.1

